src/config.rs

```diff
@@ -249,3 +279,13 @@ pub enum ServerListPing<'a> {
     /// Ignores the query and disconnects from the client.
     Ignore,
 }
+
+#[non_exhaustive]
+#[derive(Copy, Clone, Eq, PartialEq, Default)]
+pub enum ConnectionMode {
+    #[default]
+    Online,
+    Offline,
+    Bungeecord,
+    Velocity,
```

Comment:
Can we attach the secret to the `Velocity` variant of this enum? Also I've made it a heap allocated string.
```suggestion
    Velocity {
        // TODO: docs
        secret: String,
    }
```